core/src/identity/ed25519.rs

```diff
@@ -102,9 +102,19 @@ impl From<SecretKey> for Keypair {
 }
 
 /// An Ed25519 public key.
-#[derive(PartialEq, Eq, Debug, Clone)]
+#[derive(PartialEq, Eq, Clone)]
 pub struct PublicKey(ed25519::PublicKey);
 
+impl fmt::Debug for PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("PublicKey(compressed): ")?;
+        for byte in self.0.as_bytes() {
```

Comment:
Not sure whether this is relevant in the first place, but would it not be a bit more efficient to write the bytes slice to the formatter in one go, instead of one byte at a time?

Reply:
If you mean formatting the entire `&[u8]` slice with "{:x?}", that works, it is just more noisy as you get "[2a, 1b, ..., 9f]" rather than "2a1b9f", i.e. it is roughly twice as long, but I don't mind if you think it is preferable. Or did you have something else in mind?

Reply:
I somehow thought there is a formatter in `std` that would output `2a1b...9f` directly, which turns out to be a wrong assumption. Thanks for clarifying. I am in favor of merging this pull request as-is.